### PR TITLE
Fix some load/save issues with Triggers

### PR DESCRIPTION
**Pull request by zzam**
Fix some load/save issues with triggers
Fixes:
* Load will create the original number of triggers (not doubled)
* Load: No bad memory access checking for removed triggers
* Load: removed triggers are now really removed
* Save: The removed-status is correctly saved even if it was from loading a file

### CODE REVIEW

File: src/game/trigger.cpp [hunk 1]
```diff
@@ -461,10 +474,20 @@ static int CclSetActiveTriggers(lua_State *l)
 {
 	const int args = lua_gettop(l);
 
+	lua_getglobal(Lua, "_triggers_");
```
Comment (Jarod42) [inline src/game/trigger.cpp:477]:
```suggestion
	lua_getglobal(l, "_triggers_");
```

File: src/game/trigger.cpp [hunk 2]
```diff
@@ -461,10 +474,20 @@ static int CclSetActiveTriggers(lua_State *l)
...
 	ActiveTriggers.resize(args);
 	for (int j = 0; j < args; ++j) {
 		ActiveTriggers[j] = LuaToBoolean(l, j + 1);
+		if (j < triggerCount && !ActiveTriggers[j])
+		{
+			TriggerRemoveTrigger(j * 2);
+		}
 	}
+
+	lua_pop(Lua, 1);
```
Comment (Jarod42) [inline src/game/trigger.cpp:489]:
```suggestion
	lua_pop(l, 1);
```
even if `l` is always `Lua` in our code.

File: src/game/trigger.cpp [hunk 3]
```diff
@@ -391,6 +391,19 @@ void ActionStopTimer()
 	GameTimer.Running = false;
 }
 
+/**
+**  Remove a trigger
+**
+**  @param trig  Current trigger
+*/
+static void TriggerRemoveTrigger(int trig)
```
Comment (Jarod42) [inline src/game/trigger.cpp:399]: Not your code, but using `lua_State* l` would avoid global.
Add comment that top stack should be the trigger table.

File: src/game/trigger.cpp (hunk 1 above)
Comment (zzam) [reply src/game/trigger.cpp]: fixed

File: src/game/trigger.cpp (hunk 2 above)
Comment (zzam) [reply src/game/trigger.cpp]: fixed

File: src/game/trigger.cpp (hunk 3 above)
Comment (zzam) [reply src/game/trigger.cpp]: fixed

**Comment by zzam** [comment]
I will do further changes to triggers.
Is it possible to use the LuaCallback from https://github.com/Wargus/stratagus/pull/624 also for triggers?

**Review by Jarod42 — APPROVED** [review]

**Pull request merged**